Bug: Cannot declare nil as a function return type

## Description

Functions cannot declare `-> nil` as a return type, even though `nil` is a valid value in the language.

## Reproduction

```ez
import @std
using std

do main() {
    temp result = returnsNil()
    println("result: ${result}")
    println("is nil: ${result == nil}")
}

do returnsNil() -> nil {
    return nil
}
```

## Error

```
error[E2015]: expected return type after '->', got NIL instead
  --> bug_hunt.ez:10:18
   |
10 | do returnsNil() -> nil {
   |                  ^^ expected return type after arrow
```

## Expected Behavior

The parser should accept `nil` as a valid return type for functions that explicitly return `nil`.

## Additional Context

This was discovered during edge case testing. Since `nil` is a first-class value in EZ, it should be allowed as an explicit return type annotation.

--- a/integration-tests/pass/core/functions.ez
+++ b/integration-tests/pass/core/functions.ez
@@ -212,6 +212,29 @@ do main() {
         failed += 1
     }
 
+    // ==================== NIL RETURN TYPE ====================
+    println("  -- Nil Return Type --")
+
+    // Test: function with -> nil return type
+    temp nilResult = returnsNil()
+    if nilResult == nil {
+        println("  [PASS] returnsNil() -> nil returns nil")
+        passed += 1
+    } otherwise {
+        println("  [FAIL] returnsNil(): expected nil")
+        failed += 1
+    }
+
+    // Test: function with multiple return types including nil
+    temp val, err = getValueOrNil()
+    if val == 42 && err == nil {
+        println("  [PASS] getValueOrNil() -> (int, nil) works")
+        passed += 1
+    } otherwise {
+        println("  [FAIL] getValueOrNil(): val=${val}, err=${err}")
+        failed += 1
+    }
+
     // Summary
     println("")
     println("Results: ${passed} passed, ${failed} failed")
@@ -307,4 +330,12 @@ do add_score(&state GameState, points int) {
 
 do lose_life(&state GameState) {
     state.lives = state.lives - 1
+}
+
+do returnsNil() -> nil {
+    return nil
+}
+
+do getValueOrNil() -> (int, nil) {
+    return 42, nil
 }

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -1632,6 +1632,9 @@ func (p *Parser) parseFunctionDeclarationWithAttrs(attrs []*Attribute) *Function
 				return nil
 			}
 			stmt.ReturnTypes = []string{typeName}
+		} else if p.currentTokenMatches(NIL) {
+			// Single return type: nil
+			stmt.ReturnTypes = []string{"nil"}
 		} else {
 			// Invalid token after arrow
 			msg := fmt.Sprintf("expected return type after '->', got %s instead", p.currentToken.Type)
@@ -1871,13 +1874,16 @@ func (p *Parser) parseReturnTypes() []string {
 	p.nextToken() // move past (
 
 	for !p.currentTokenMatches(RPAREN) {
-		// Parse type name (can be IDENT or array type starting with LBRACKET)
+		// Parse type name (can be IDENT, array type starting with LBRACKET, or nil)
 		if p.currentTokenMatches(IDENT) || p.currentTokenMatches(LBRACKET) {
 			typeName := p.parseTypeName()
 			if typeName == "" {
 				return nil
 			}
 			types = append(types, typeName)
+		} else if p.currentTokenMatches(NIL) {
+			// nil return type
+			types = append(types, "nil")
 		} else {
 			msg := fmt.Sprintf("expected type name in return types, got %s", p.currentToken.Type)
 			p.addEZError(errors.E2002, msg, p.currentToken)

